framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -5007,7 +5008,9 @@ private void copy(AnnotatedDeclaredType uncapturedType, AnnotatedDeclaredType ca
   @SuppressWarnings("interning:not.interned") // must be the same object from collection
   private AnnotatedTypeVariable doesNotContainOthers(
       Collection<? extends AnnotatedTypeVariable> collection) {
+    AnnotatedTypeVariable first = null;
     for (AnnotatedTypeVariable candidate : collection) {
+      first = candidate;
```

Comment:
The current code returns the last, not first, candidate if no other matches.
I think this line should be `if (first == null) { first = candidate; }`.